src/main/java/org/broadinstitute/hellbender/utils/config/GATKConfig.java

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.hellbender.utils.config;
+
+import org.aeonbits.owner.Accessible;
+import org.aeonbits.owner.Mutable;
+import org.aeonbits.owner.Config.LoadPolicy;
+import org.aeonbits.owner.Config.LoadType;
+import org.aeonbits.owner.Config.Sources;
+
+import java.util.List;
+
+/**
+ * Configuration file for GATK options.
+ */
+@LoadPolicy(LoadType.MERGE)
+@Sources({
+        "file:${" + GATKConfig.CONFIG_FILE_VARIABLE_NAME + "}",                         // Variable for file loading
+        "file:GATKConfig.properties",                                                   // Default path
+        "classpath:org/broadinstitute/hellbender/utils/config/GATKConfig.properties"    // Class path
+})
+public interface GATKConfig extends Mutable, Accessible {
```

Comment:
Thanks for the answer, @jonn-smith. Some more details:

1. What I mean with this one is that I do not want to expose all the configuration parameters to the user, so I would like to provide defaults that are not overriden by a config-file. Is that possible?
2. I think that in `ConfigCache.getOrCreate` I can link both `GATKConfig.class` and `ExtendedGATKConfig.class` to the same instance of `ExtendedGATKConfig` if it is extending the previous. If it is the first thing that I do (statically), I will only have one cached config instance for retrieve by the GATK code and my own code. My question here how OWNER cope with this extended classes, and if this could raise an error if done.
3. My question here is not from the final user point of view, but as a developer using GATK as a framework. I do not want to be able to provide a config file to the user yet (maybe in the future, but it will require also to hide some of the values from the final user), but override some of the defaults provided by GATK. In addition, I would like to have flexibility on how to include the codecs, as I pointed in a previous comment: I would like to use the default codecs plus the ones implemented in my code, and do not use codecs that conflicts. This requires a finer control over how the properties are merged, and it is not clear for me how can this be done.